generator/internal/genclient/translator/openapi/openapi_test.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openapi
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/googleapis/google-cloud-rust/generator/internal/genclient"
+)
+
+func TestMakeAPI(t *testing.T) {
+	tDir := t.TempDir()
+
+	contents := []byte(testDocument)
+	translator, err := NewTranslator(contents, &Options{
+		Language:    "rust",
+		OutDir:      tDir,
+		TemplateDir: "not used",
+	})
+	if err != nil {
+		t.Errorf("Error in NewTranslator() %q", err)
+	}
+
+	api, err := translator.makeAPI()
+	if err != nil {
+		t.Errorf("Error in makeAPI() %q", err)
+	}
+
+	checkMessage(t, *api.Messages[0], genclient.Message{
+		Documentation: "A resource that represents a Google Cloud location.",
+		Name:          "Location",
+		Fields: []*genclient.Field{
+			{Name: "name",
```

Comment:
nit: ident before Name here and below

Reply:
`go fmt` is removing any indentation, but moving to the next line seems to work.